StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/FileHeaderCodeFixProvider.cs

```diff
@@ -56,48 +57,81 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             var settings = document.Project.AnalyzerOptions.GetStyleCopSettings();
 
             var fileHeader = FileHeaderHelpers.ParseFileHeader(root);
-            var newSyntaxRoot = fileHeader.IsMissing ? AddHeader(document, root, document.Name, settings) : ReplaceHeader(document, root, settings);
+            var xmlFileHeader = FileHeaderHelpers.ParseXmlFileHeader(root);
+            var newSyntaxRoot = fileHeader.IsMissing ? AddHeader(document, root, document.Name, settings) : ReplaceHeader(document, root, settings, xmlFileHeader.IsMalformed);
 
             return document.WithSyntaxRoot(newSyntaxRoot);
         }
 
-        private static SyntaxNode ReplaceHeader(Document document, SyntaxNode root, StyleCopSettings settings)
+        private static SyntaxNode ReplaceHeader(Document document, SyntaxNode root, StyleCopSettings settings, bool isMalformedHeader)
         {
+            // If the header is well formed Xml then we parse out the copyright otherwise
             // Skip single line comments, whitespace, and end of line trivia until a blank line is encountered.
             SyntaxTriviaList trivia = root.GetLeadingTrivia();
-
             bool onBlankLine = false;
-            while (trivia.Any())
+            bool inCopyright = isMalformedHeader;
+            int? copyrightTriviaIndex = null;
+            var removalList = new System.Collections.Generic.List<int>();
+
+            // Need to do this with index so we get the line endings correct.
+            for (int i = 0; i < trivia.Count; i++)
             {
+                var triviaLine = trivia[i];
                 bool done = false;
-                switch (trivia[0].Kind())
+                switch (triviaLine.Kind())
                 {
-                case SyntaxKind.SingleLineCommentTrivia:
-                    trivia = trivia.RemoveAt(0);
-                    onBlankLine = false;
-                    break;
-
-                case SyntaxKind.WhitespaceTrivia:
-                    trivia = trivia.RemoveAt(0);
-                    break;
-
-                case SyntaxKind.EndOfLineTrivia:
-                    trivia = trivia.RemoveAt(0);
-
-                    if (onBlankLine)
-                    {
+                    case SyntaxKind.SingleLineCommentTrivia:
+                        if (!isMalformedHeader)
+                        {
+                            var openingTag = triviaLine.ToFullString().Contains("<copyright ");
+                            var closingTag = triviaLine.ToFullString().Contains("</copyright>") ||
+                                (openingTag && triviaLine.ToFullString().Trim().EndsWith("/>"));
+                            if (openingTag)
+                            {
+                                inCopyright = !closingTag;
+                                copyrightTriviaIndex = i;
+                            }
+                            else
+                            {
+                                if (inCopyright)
```

Comment:
:thought_balloon: Reformatting to `else if (inCopyright)` would improve readability